src/parser/grammar.lalrpop

```diff
@@ -821,8 +828,14 @@ InfixExpr: UniTerm = {
     InfixLazyBOpApp<InfixLazyBOp10, InfixExpr, InfixExpr>,
 
     #[precedence(level="11")] #[assoc(side="right")]
-    <s: AsType<InfixExpr>> "->" <t: AsType<InfixExpr>> =>
-        UniTerm::from(Types(TypeF::Arrow(Box::new(s), Box::new(t)))),
+    <l: @L> <s: AsType<InfixExpr>> "->" <t: AsType<InfixExpr>> <r: @R> =>
```

Comment:
Maybe you can even use `WithPos` in the `AsType` macro, and call it a day?

Reply:
The `AsType` rule already uses the `WithPos` macro